src/geode/inspector/topology/brep_topology.cpp

```diff
@@ -691,96 +315,222 @@ namespace geode
         BRepTopologyInspector::invalid_components_topology_unique_vertices()
             const
     {
-        return impl_->invalid_components_topology_unique_vertices();
+        std::vector< index_t > invalid_unique_vertices;
+        for( const auto unique_vertex_id : Range{ brep_.nb_unique_vertices() } )
+        {
+            if( !brep_corner_topology_is_valid( unique_vertex_id )
+                || !brep_vertex_lines_topology_is_valid( unique_vertex_id )
+                || !brep_vertex_surfaces_topology_is_valid( unique_vertex_id )
+                || !brep_vertex_blocks_topology_is_valid( unique_vertex_id ) )
+            {
+                invalid_unique_vertices.push_back( unique_vertex_id );
+            }
+        }
+        return invalid_unique_vertices;
     }
 
     std::vector< index_t >
         BRepTopologyInspector::multiple_corners_unique_vertices() const
     {
-        return impl_->multiple_corners_unique_vertices();
+        std::vector< index_t > invalid_unique_vertices;
+        for( const auto unique_vertex_id : Range{ brep_.nb_unique_vertices() } )
+        {
+            if( unique_vertex_has_multiple_corners( unique_vertex_id ) )
+            {
+                invalid_unique_vertices.push_back( unique_vertex_id );
+            }
+        }
+        return invalid_unique_vertices;
     }
 
     std::vector< index_t >
         BRepTopologyInspector::multiple_internals_corner_vertices() const
     {
-        return impl_->multiple_internals_corner_vertices();
+        std::vector< index_t > invalid_unique_vertices;
+        for( const auto unique_vertex_id : Range{ brep_.nb_unique_vertices() } )
```

Comment:
There are a lot of these methods that just do a for and call the mother classes. Should this loop be implemented in this class or in the mixin/mother class? I guess the same code will be for the section, is there a way to mutualize this loop?

Reply:
On a larger comment, is B-RepTopology should have methods or should we only call the mixin methods directly?